tensorboard/components_polymer3/tf_card_heading/tf-card-heading.ts

```diff
@@ -0,0 +1,186 @@
+ *     card; if left unset, no border will be displayed.
+ *   - Any children of this component will be rendered below the above
+ *     information, and can be used to display (for example) some small
+ *     control widgets.
+ */
+@customElement('tf-card-heading')
+class TfCardHeading extends PolymerElement {
+  static readonly template = html`
+    <div class="container">
+      <figcaption class="content">
+        <div class="heading-row">
+          <template is="dom-if" if="[[_nameLabel]]">
+            <div itemprop="name" class="heading-label name">
+              [[_nameLabel]]
+            </div>
+          </template>
+          <template is="dom-if" if="[[run]]">
+            <!-- Extra wrapping span needed to avoid flexbox blockification. -->
+            <!-- (see flexbox spec, section 4 "Flex Items") -->
+            <span>
+              <span
+                itemprop="run"
+                id="heading-run"
+                class="heading-label heading-right run"
+                >[[run]]</span
+              >
+            </span>
+          </template>
+        </div>
+        <template is="dom-if" if="[[_tagLabel]]">
+          <div class="heading-row">
+            <div class="heading-label">
+              tag: <span itemprop="tag">[[_tagLabel]]</span>
+            </div>
+          </div>
+        </template>
+        <slot></slot>
+      </figcaption>
+      <template is="dom-if" if="[[description]]">
+        <paper-icon-button
+          icon="info"
+          on-tap="_toggleDescriptionDialog"
+          title="Show summary description"
+        ></paper-icon-button>
+      </template>
+      <paper-dialog
+        id="descriptionDialog"
+        no-overlap=""
+        horizontal-align="auto"
+        vertical-align="auto"
+      >
+        <paper-dialog-scrollable>
+          <tf-markdown-view html="[[description]]"></tf-markdown-view>
+        </paper-dialog-scrollable>
+      </paper-dialog>
+    </div>
+    <style include="tf-card-heading-style">
+      .container {
+        display: flex;
+      }
+      .content {
+        font-size: 12px;
+        flex-grow: 1;
+      }
+      .name {
+        font-size: 14px;
+      }
+      .run {
+        font-size: 11px;
+        width: auto;
+        border-radius: 3px;
+        font-weight: bold;
+        padding: 1px 4px 2px;
+      }
+      paper-icon-button {
+        flex-grow: 0;
+      }
+      paper-dialog-scrollable {
+        max-width: 640px;
+      }
+      #heading-run {
+        background: var(--tf-card-heading-background-color);
+        color: var(--tf-card-heading-color);
+      }
+    </style>
+  `;
+
+  @property({type: String})
+  displayName: string = null;
+
+  @property({type: String})
+  tag: string = null;
+
+  @property({type: String})
+  run: string = null;
+
+  @property({type: String})
+  description: string = null;
+
+  @property({type: String})
+  color: string = null;
+
+  @property({
```

Comment:
Just for my understanding - it seems like in the Polymer 3 style this might be more idiomatically converted to also be an `@computed` property on `_computeRunBackground()` (and likewise for `_runColor` below)?  (It sounds like that supports `observer` still, but at any rate it also seems like maybe `_updateHeadingStyle()` would be more idiomatic as an `@observer` anyway?)

I'm totally fine if our preference is to stick with the minimum working conversion and not bother trying to optimize, just curious for my own information whether that's an accurate reading here?

Reply:
I forgot the condition in which the tool automatically concerts that but something was not deemed safe or unhandled. 

I am going for minimal human intervention at this point since the decorator syntax is a mere sugar on top of this `property` decorator. When converting yourself, try to not make mechanical changes but do look out for incorrect typing/wrong conversions.